vita3k/gui/src/controls_dialog.cpp

```diff
@@ -15,45 +15,91 @@
+    remapper_button(gui, host, &host.cfg.keyboard_leftstick_down, "Left stick down", &ImVec2(7.0f, 7.0f));
+    remapper_button(gui, host, &host.cfg.keyboard_leftstick_right, "Left stick right", &ImVec2(7.0f, 7.0f));
+    remapper_button(gui, host, &host.cfg.keyboard_leftstick_left, "Left stick left", &ImVec2(7.0f, 7.0f));
+    remapper_button(gui, host, &host.cfg.keyboard_rightstick_up, "Right stick up", &ImVec2(7.0f, 7.0f));
+    remapper_button(gui, host, &host.cfg.keyboard_rightstick_down, "Right stick down", &ImVec2(7.0f, 7.0f));
+    remapper_button(gui, host, &host.cfg.keyboard_rightstick_right, "Right stick right", &ImVec2(0.3f, 0.3f));
+    remapper_button(gui, host, &host.cfg.keyboard_rightstick_left, "Right stick left", &ImVec2(7.0f, 7.0f));
+    remapper_button(gui, host, &host.cfg.keyboard_button_up, "D-pad up", &ImVec2(7.0f, 7.0f));
+    remapper_button(gui, host, &host.cfg.keyboard_button_down, "D-pad down", &ImVec2(7.0f, 7.0f));
+    remapper_button(gui, host, &host.cfg.keyboard_button_right, "D-pad right", &ImVec2(7.0f, 7.0f));
+    remapper_button(gui, host, &host.cfg.keyboard_button_left, "D-pad left", &ImVec2(7.0f, 7.0f));
     if (host.cfg.pstv_mode) {
-        ImGui::Text("%-16s    %-16s", "L1 button", "Q");
-        ImGui::Text("%-16s    %-16s", "R1 button", "E");
-        ImGui::Text("%-16s    %-16s", "L2 button", "U");
-        ImGui::Text("%-16s    %-16s", "R2 button", "O");
-        ImGui::Text("%-16s    %-16s", "L3 button", "F");
-        ImGui::Text("%-16s    %-16s", "R3 button", "H");
+        remapper_button(gui, host, &host.cfg.keyboard_button_l1, "L1 button", &ImVec2(7.0f, 7.0f));
+        remapper_button(gui, host, &host.cfg.keyboard_button_r1, "R2 button", &ImVec2(7.0f, 7.0f));
+        remapper_button(gui, host, &host.cfg.keyboard_button_l2, "L2 button", &ImVec2(7.0f, 7.0f));
+        remapper_button(gui, host, &host.cfg.keyboard_button_r2, "R2 button", &ImVec2(7.0f, 7.0f));
+        remapper_button(gui, host, &host.cfg.keyboard_button_l3, "L3 button", &ImVec2(7.0f, 7.0f));
+        remapper_button(gui, host, &host.cfg.keyboard_button_r3, "R3 button", &ImVec2(7.0f, 7.0f));
     } else {
-        ImGui::Text("%-16s    %-16s", "L button", "Q");
-        ImGui::Text("%-16s    %-16s", "R button", "E");
+        remapper_button(gui, host, &host.cfg.keyboard_button_l1, "L button", &ImVec2(7.0f, 7.0f));
+        remapper_button(gui, host, &host.cfg.keyboard_button_r1, "R button", &ImVec2(7.0f, 7.0f));
     }
-    ImGui::Text("%-16s    %-16s", "Square button", "Z");
-    ImGui::Text("%-16s    %-16s", "X button", "X");
-    ImGui::Text("%-16s    %-16s", "Circle button", "C");
-    ImGui::Text("%-16s    %-16s", "Triangle button", "V");
-    ImGui::Text("%-16s    %-16s", "Start button", "Enter");
-    ImGui::Text("%-16s    %-16s", "Select button", "Right shift");
+    remapper_button(gui, host, &host.cfg.keyboard_button_square, "Square button", &ImVec2(7.0f, 7.0f));
```

Comment:
Your build is failing because of `&ImVec2(7.0f, 7.0f)`, surprised mac and windows actually compiles this XD 

Think about what the `&` operator really does- It takes an address of something on stack, usually a variable. However, with `ImVec2(7.0f, 7.0f)` you aren't really creating a variable at all, it just briefly creates an `ImVec2` struct until the function is called. It's likely that once execution reaches the function, the memory that `&ImVec2(7.0f, 7.0f)` points to is already freed and therefore invalid. `new ImVec2(7.0f, 7.0f)` would solve this issue but would cause a leak instead... so what do we do???

Well it looks like you always pass exactly the same ImVec2, so you could probably drop the argument entirely and just always pass the same vector to ImGui in `remapper_button`.

Another way you could accomplish this is by declaring a variable with the value and passing an address to that variable:
```
ImVec2 dummy_size = ImVec2(7.0f, 7.0f);
...
remapper_button(..., &dummy_size);
```

You could also change the argument type to `ImVec2 dummy_size` or `const ImVec2 &dummy_size` (better), and that will let you just pass the constructor like this:
```
remapper_button(..., ImVec2(7.0f, 7.0f));
```

Reply:
Thank you